crates/tessera-backbone/src/domain/secret/mod.rs

```diff
@@ -13,10 +15,42 @@ pub(crate) struct SecretEntry {
     pub writer_policy_ids: Vec<Ulid>,
 }
 
+impl From<(secret_metadata::Model, Vec<applied_policy::Model>)> for SecretEntry {
+    fn from((metadata, applied_policies): (secret_metadata::Model, Vec<applied_policy::Model>)) -> Self {
+        let mut reader_policy_ids: Vec<Ulid> = vec![];
+        let mut writer_policy_ids: Vec<Ulid> = vec![];
+
+        for applied_policy in applied_policies {
+            match applied_policy.r#type {
+                applied_policy::PolicyApplicationType::Read => reader_policy_ids.push(applied_policy.id.inner()),
+                applied_policy::PolicyApplicationType::Write => writer_policy_ids.push(applied_policy.id.inner()),
+            }
+        }
+
+        SecretEntry { key: metadata.key, path: metadata.path, reader_policy_ids, writer_policy_ids }
+    }
+}
+
 #[cfg_attr(test, automock)]
 #[async_trait]
 pub(crate) trait SecretService {
     async fn list(&self, transaction: &DatabaseTransaction, path_prefix: &str) -> Result<Vec<SecretEntry>>;
+
+    async fn get(&self, transaction: &DatabaseTransaction, secret_identifier: &str) -> Result<SecretEntry>;
+}
+
+lazy_static! {
+    static ref IDENTIFIER_PATTERN: Regex =
+        Regex::new(r"(.*)/([^/]+)").expect("IDENTIFIER_PATTERN should be compiled successfully");
```

Comment:
The current regex pattern `(.*)/([^/]+)` may have some potential issues: 
1. It could match strings without a leading slash, which may not be the intended behavior for path identifiers.  
2. It doesn't account for empty path segments (e.g., `/path//path2`), which could lead to unexpected matches.  
3. [ReDoS](https://devina.io/redos-checker)

To address these concerns, I suggest considering the following regex pattern:  

```
^\/(?:[^/]+\/)*[^/]+$  
```

Reply:
I will apply it as follows, since it need to capture the path and key.

`^((?:/[^/]+)*)/([^/]+)$`